### PR TITLE
ref: split incident and alert_rule models into separate files

### PR DESCRIPTION
**Pull request by nhsiehgit**
currently the incidents/models.py file is nearly 1k lines long and the file name `model` does not do much to tell us what it's contents are

PR to split up the file names and directory

edit:

oh lawd. i swear most of these are modifying imports now that the directory structure has been modified

Resolves https://github.com/getsentry/getsentry/pull/13170
Requires https://github.com/getsentry/getsentry/pull/13170

### CODE REVIEW

File: src/sentry/incidents/models/incident.py [hunk 1]
```diff
@@ -0,0 +1,395 @@
+from __future__ import annotations
```
Comment (nhsiehgit) [inline src/sentry/incidents/models/incident.py:1]: THIS IS THE MAIN CHANGE IN THIS PR

File: src/sentry/incidents/models/alert_rule.py [hunk 2]
```diff
@@ -6,44 +6,39 @@
 from datetime import timedelta
```
Comment (nhsiehgit) [inline src/sentry/incidents/models/alert_rule.py:6]: THIS IS THE OTHER MAIN CHANGE OF THIS PR

**Review by wedamija — APPROVED** [review]
I skimmed over this with `git diff --color-moved=dimmed_zebra HEAD^` and it looks fine, hard to thoroughly review this but if ci passed it should be fine.

You should probably split up `test_models.py` in the same way

**Comment by JoshFerge** [comment]
do we need to worry about errors during deploy similar to the renaming we did for releases?

**Comment by wedamija** [comment]
> do we need to worry about errors during deploy similar to the renaming we did for releases?

Only if we're using `buffer_incr` to increment a field on the models. Double checked the codebase and we're fine here.

**Review by saponifi3d — APPROVED** [review]
looks good and makes sense to me. 🎉 - just want to confirm we have lint setup to validate / sort imports?

File: src/sentry/incidents/models/incident.py [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+    DETECTED = 4
+
+
+@region_silo_only_model
+class IncidentActivity(Model):
+    __relocation_scope__ = RelocationScope.Organization
+
+    incident = FlexibleForeignKey("sentry.Incident")
+    user_id = HybridCloudForeignKey(settings.AUTH_USER_MODEL, on_delete="CASCADE", null=True)
+    type: models.Field[int | IncidentActivityType, int] = models.IntegerField()
```
Comment (nhsiehgit) [inline src/sentry/incidents/models/incident.py:294]: 🤔 

Why would this line be perfectly fine in the singular `models` file - however now that it's split out into a separate file, mypy i snow angry at the type???

I'm seeing

```
error: Incompatible types in assignment (expression has type "IntegerField[float | int | str | Combinable, int]", variable has type "Field[int | IncidentActivityType, int]")  [assignment]
```

and i'm not entirely sure how to get around it...
importing `Combinable` seems to be having issues as well 🤔

Comment (wedamija) [reply src/sentry/incidents/models/incident.py:294]: Not sure if related, but you're missing an `__init__.py` from your models folder.

Comment (JoshFerge) [reply src/sentry/incidents/models/incident.py]: yeah, i think i've seen this before with missing __init__'s

**Pull request merged**